app/views/service_worker/index.js.erb

```diff
@@ -1,91 +1,21 @@
-        // response isn't fulfilled from the HTTP cache; i.e., it will be from
-        // the network.
-        const offlineCache = await caches.open(OFFLINE_NAME);
-        await offlineCache.add(new Request(OFFLINE_URL, { cache: "reload" }));
-      })()
-    );
-    // Force the waiting service worker to become the active service worker.
+<% unless Rails.env.test? %>
+  self.addEventListener('install', function(e) {
     self.skipWaiting();
   });
 
-  self.addEventListener("activate", (event) => {
-    event.waitUntil(
-      (async () => {
-        // Enable navigation preload if it's supported.
-        // See https://developers.google.com/web/updates/2017/02/navigation-preload
-        if ("navigationPreload" in self.registration) {
-          await self.registration.navigationPreload.enable();
-        }
-      })()
-    );
-
-    // Tell the active service worker to take control of the page immediately.
-    self.clients.claim();
-  });
-
-  function includesUnsupportedPath(url) {
-    return [
-      '/%F0%9F%92%B8',         // 💸 (hiring)
-      '/admin',                // Don't run on admin dashboard.
-      '/new',                  // Don't run on editor.
-      '/enter',                // Don't run on sign in page.
-      '/oauth/',               // Skip oauth apps
-      '/onboarding',           // Don't run on onboarding.
-      '/shop',                 // Don't run on Shop
-      '/sidekiq',              // Skip for Sidekiq dashboard
-      '/users/auth',           // Don't run on authentication.
-      '/users/sign_in',        // Don't run on sign in page.
-      '/welcome',              // Don't run on welcome reroutes.
-    ].some(path => url.includes(path))
-  }
-
-  self.addEventListener("fetch", (event) => {
-    // We only want to call event.respondWith() if this is a navigation request
-    // for an HTML page with a few exceptions.
-    if (event.request.mode === "navigate" && !includesUnsupportedPath(event.request.url)) {
-      event.respondWith(
-        (async () => {
-          try {
-            // First, try to use the navigation preload response if it's supported.
-            const preloadResponse = await event.preloadResponse;
-            if (preloadResponse) {
-              return preloadResponse;
-            }
-
-            // Always try the network first.
-            const networkResponse = await fetch(event.request);
-            return networkResponse;
-          } catch (error) {
-            // catch is only triggered if an exception is thrown, which is likely
-            // due to a network error.
-            // If fetch() returns a valid HTTP response with a response code in
-            // the 4xx or 5xx range, the catch() will NOT be called.
-            console.log("Fetch failed; returning offline page instead.", error);
-
-            const cache = await caches.open(OFFLINE_NAME);
-            const cachedResponse = await cache.match(OFFLINE_URL);
-            return cachedResponse;
```

Comment:
Does this mean we will no longer have the fallback offline (doodle) page or will it continue to work but just not triggered from here?

Reply:
@fdoxyz, the offline page was a feature of the service worker, so the only way you'd be able to access the page now is by going directly to e.g. http://localhost:3000/offline.html

Reply:
Should we also remove that file while we're here? Could always recover it later if we want to bring it back.

Reply:
Sounds good @Zhao-Andy. Will do.

Reply:
It's been removed. cc: @fdoxyz.